lib/tesla/adapter/gun.ex

```diff
@@ -170,6 +170,7 @@ if Code.ensure_loaded?(:gun) do
       end
     end
 
+    @dialyzer [{:nowarn_function, open_conn: 2}, :no_match]
```

Comment:
Why do we need this annotation?

Reply:
Dialyzer complains about some inconsistencies when not using the `gun` from master branch, such as `origin_scheme`, `tls_opts`, or `tcp_opts` not found on `:gun.opts()` type when opening a gun connection.

Reply:
Yeah, these warnings are rather annoying 😅 Thanks for the fix